test-dev/libxmp_fuzz.c

```diff
@@ -38,21 +81,206 @@ int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
 		xmp_set_player(opaque, XMP_PLAYER_INTERP, interp);
 
 		/* TODO Saga Musix also recommends performing different types of seeking. */
-		for (i = 0; i < 64; i++)
-			xmp_play_frame(opaque);
+		for (i = 0; i < frames_to_play; i++)
+		{
+			int r = xmp_play_frame(opaque);
+			if (r != 0)
+				play_error = r;
+		}
 
 		xmp_release_module(opaque);
 	}
 
 	/* Fuzz depackers. */
-	f = fmemopen((void *)data, size, "rb");
 	if (f != NULL)
 	{
 		struct xmp_test_info info;
-		xmp_test_module_from_file(f, &info);
+		test_error = xmp_test_module_from_file(f, &info);
+	}
+
+#ifndef LIBXMP_LIBFUZZER
+	if (!quiet)
+	{
+		/* Print and log status */
+		total++;
+		if (load_error <= 0 && load_error > -256)
+			status_count[-load_error]++;
+		else
+			status_unkn++;
+
+		O_(" load:%d", load_error);
+		if (f != NULL)
+			O_(" test:%d", test_error);
+		if (load_error == 0)
+			O_(" play:%d", play_error);
+	}
+#endif
+	return 0;
+}
+
+#ifdef LIBXMP_LIBFUZZER
+
+/**
+ * libFuzzer wrapper function.
+ */
+#ifdef __cplusplus
+extern "C"
+#endif
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+	xmp_context opaque = xmp_create_context();
+	FILE *f = NULL;
+
+#ifdef HAVE_FMEMOPEN
+	f = fmemopen((void *)data, size, "rb");
```

Comment:
What happens if fmemopen isn't available, i.e. f == NULL, here? (Similar down below.)

Reply:
`libxmp_test_function` guards against `NULL` from this and then it doesn't get closed. The only reason this is passed to `libxmp_test_function` in the first place is that I was considering having the command line version pass the original `FILE` directly, and then I changed my mind... I'll just move this into `libxmp_test_function`.

Reply:
Okay, I did this, and also moved a few things around so all the includes and static variables are at the top.

Reply:
OK, will merge as soon as CI completes.